test/integration/integration_test.sh

```diff
@@ -3,192 +3,175 @@
 # AzMiTool Integration tests
 # It requires Bash Testing Framework
 
+
+#
 # setup variables
+#
+
+# this script is called from testing framework script, therefore arguments are shifted
+STORAGEACCOUNTNAME=$2
+identity=$3
+
+
 export DEBIAN_FRONTEND=noninteractive
 PACKAGENAME=azmi
 PACKAGEFILE=/tmp/azmiX.deb
+declare -a subCommands=("gettoken" "getblob" "getblobs" "setblob" "listblobs")
+identity_foreign=017dc05c-4d12-4ac2-b5f8-5e239dc8bc54
 
-# define function(s)
-function install_azmi() { # installs/upgrades package from file if exists; otherwise APT repository
-  if $(dpkg-query --show $PACKAGENAME > /dev/null 2>&1); then
-    EVENT="Upgrade"
-  else
-    EVENT="Install"
-  fi
-
-  if [ -f $PACKAGEFILE ]; then
-      test "$EVENT $PACKAGENAME package from file" assert.Success "dpkg -i $PACKAGEFILE"
-  else
-      test "$EVENT $PACKAGENAME package from repository" assert.Success "apt --assume-yes install $PACKAGENAME"
-  fi  
-  dpkg-query --showformat='  - ${Package} ${Version} installed\n' --show $PACKAGENAME
-}
+# calculated variables
 
+
+#
 # start testing
+#
+
 testing start "$PACKAGENAME"
 testing class "package"
 test "Install fake package should fail" assert.Fail "apt --assume-yes install somenonexistingpackage"
 
 # dependencies installed?
 test "Check all dependencies are installed" assert.Success "dpkg -s libstdc++6"
 
-install_azmi
+test "Install $PACKAGENAME package from file" assert.Success "dpkg -i $PACKAGEFILE"
```

Comment:
Lost possible upgrade scenario.

Reply:
Thanks for catching this. It's a topic I wanted to discuss. Taking it 1-1